logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/debug/SendBrokenAssetMessageUseCaseImpl.kt

```diff
@@ -0,0 +1,228 @@
+     * clients itself.
+     *
+     * In contrast to SendAssetMessageUseCase this debug function does not persist the message.
+     *
+     * @param conversationId the id of the conversation where the asset wants to be sent
+     * @param assetDataPath the raw data of the asset to be uploaded to the backend and sent to the given conversation
+     * @param assetDataSize the size of the original asset file
+     * @param assetName the name of the original asset file
+     * @param assetMimeType the type of the asset file
+     * @param brokenState the type of manipulation
+     * @return an [SendBrokenAssetMessageResult] containing a [CoreFailure] in case anything goes wrong and [Unit] in case everything succeeds
+     */
+    @Suppress("LongParameterList")
+    suspend operator fun invoke(
+        conversationId: ConversationId,
+        assetDataPath: Path,
+        assetDataSize: Long,
+        assetName: String,
+        assetMimeType: String,
+        brokenState: BrokenState
+    ): SendBrokenAssetMessageResult
+}
+
+@Suppress("LongParameterList", "MaxLineLength")
+internal class SendBrokenAssetMessageUseCaseImpl(
+    private val clientRepository: ClientRepository,
+    private val assetDataSource: AssetRepository,
+    private val userRepository: UserRepository,
+    private val slowSyncRepository: SlowSyncRepository,
+    private val messageSender: MessageSender
+) : SendBrokenAssetMessageUseCase {
+    private lateinit var currentAssetMessageContent: AssetMessageMetadata
+    override suspend fun invoke(
+        conversationId: ConversationId,
+        assetDataPath: Path,
+        assetDataSize: Long,
+        assetName: String,
+        assetMimeType: String,
+        brokenState: BrokenState
+    ): SendBrokenAssetMessageResult {
+        slowSyncRepository.slowSyncStatus.first {
+            it is SlowSyncStatus.Complete
+        }
+
+        // Generate the otr asymmetric key that will be used to encrypt the data
+        val otrKey = generateRandomAES256Key()
+        currentAssetMessageContent = AssetMessageMetadata(
+            conversationId = conversationId,
+            mimeType = assetMimeType,
+            assetDataPath = assetDataPath,
+            assetDataSize = assetDataSize,
+            assetName = assetName,
+            assetWidth = 0,
+            assetHeight = 0,
+            otrKey = otrKey,
+            sha256Key = SHA256Key(ByteArray(DEFAULT_BYTE_ARRAY_SIZE)), // Sha256 will be replaced with right values after asset upload
+            assetId = UploadedAssetId("", ""), // Asset ID will be replaced with right value after asset upload
+        )
+        lateinit var message: Message.Regular
+
+        return clientRepository.currentClientId().flatMap { currentClientId ->
+            // Get my current user
+            val selfUser = userRepository.observeSelfUser().firstOrNull()
+
```

Comment:
So you need the selfUserId this should injected directly since it's available in the UserSessionScope.